src/pip/_internal/commands/cache.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import absolute_import
+
+import logging
+import os
+import textwrap
+
+from pip._internal.cli.base_command import Command
+from pip._internal.exceptions import CommandError
+from pip._internal.utils.filesystem import find_files
+
+logger = logging.getLogger(__name__)
+
+
+class CacheCommand(Command):
+    """
+    Inspect and manage pip's caches.
+
+    Subcommands:
+        info:
+            Show information about the caches.
+        list [name]:
+            List filenames of packages stored in the cache.
+        remove <pattern>:
+            Remove one or more package from the cache.
+            `pattern` can be a glob expression or a package name.
+        purge:
+            Remove all items from the cache.
+    """
+    actions = ['info', 'list', 'remove', 'purge']
+    name = 'cache'
+    usage = """
+      %prog <command>"""
+    summary = "View and manage which packages are available in pip's caches."
+
+    def __init__(self, *args, **kw):
+        super(CacheCommand, self).__init__(*args, **kw)
+
+    def run(self, options, args):
+        if not args:
+            raise CommandError('Please provide a subcommand.')
+
+        if args[0] not in self.actions:
+            raise CommandError('Invalid subcommand: %s' % args[0])
+
+        self.wheel_dir = os.path.join(options.cache_dir, 'wheels')
+
+        method = getattr(self, 'action_%s' % args[0])
+        return method(options, args[1:])
+
+    def action_info(self, options, args):
+        format_args = (options.cache_dir, len(self.find_wheels('*.whl')))
+        result = textwrap.dedent(
+            """\
+            Cache info:
+              Location: %s
+              Packages: %s""" % format_args
+        )
+        logger.info(result)
+
+    def action_list(self, options, args):
+        if args and args[0]:
+            pattern = args[0]
+        else:
+            pattern = '*'
+
+        files = self.find_wheels(pattern)
+        wheels = map(self._wheel_info, files)
```

Comment:
I'm not convinced that hiding the python, abi & platform tags (via `self._wheel_info`) is a good idea...
One use case would be to quickly identify all Python 3.5 wheels once you know you won't work with it anymore

Reply:
Looking at it again, I'm not convinced it is, either. iirc it's the only way I could get it to work nicely, but I don't recall the details of why that is. Will need to look into it more.

Reply:
@xavfernandez my latest changes make it so it just shows the entire file name, but _wheel_info still removes the directory. e.g.

```
$ python3 src/pip cache list
Current cache contents:
 - PyYAML-5.1.2-cp27-cp27mu-linux_x86_64.whl
 - PyYAML-5.1.2-cp37-cp37m-linux_x86_64.whl
 - functools32-3.2.3.post2-cp27-none-any.whl
 - mypy_extensions-0.4.2-py2.py3-none-any.whl
 - nodeenv-1.3.3-cp37-none-any.whl
 - pip-19.3.dev0-cp27-none-any.whl
 - pip-19.3.dev0-cp37-none-any.whl
 - scandir-1.10.0-cp27-cp27mu-linux_x86_64.whl
 - scripttest-1.3-cp27-none-any.whl
 - scripttest-1.3-cp37-none-any.whl
 - snowballstemmer-1.9.1-cp37-none-any.whl
$
```